### PR TITLE
Feature ETP-871: Structure for report page

### PR DESCRIPTION
**Pull request by Sankku**

### CODE REVIEW

**Comment by LuuchoRocha** [comment]
@Sankku please address the build issues

File: packages/ComponentLibrary/src/hooks/useItemType.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    if (item.windowId) {
+      return item.windowId;
+    }
+
+    if (item.tableId) {
+      return item.tableId;
+    }
+
+    return item.id;
+  }, []);
```
Comment (LuuchoRocha) [inline packages/ComponentLibrary/src/hooks/useItemType.ts:25]: please move this outside of the hook (it doesn't need to be defined as a callback inside of the hook, it's a regular helper)

File: packages/ComponentLibrary/src/hooks/useItemType.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          }
+          break;
+        default:
+          console.warn(`Unhandled item type: ${item.type}`);
+      }
+    },
+    [onWindowClick, onReportClick, onProcessClick, getReportRecordId],
+  );
+
+  return { handleItemClick };
```
Comment (LuuchoRocha) [inline packages/ComponentLibrary/src/hooks/useItemType.ts:53]:
```suggestion
  return handleItemClick;
```

File: packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx [hunk 3]
```diff
@@ -29,6 +30,12 @@ const DrawerSection: React.FC<DrawerSectionProps> = ({
 
   const popperOpen = Boolean(anchorEl);
 
+  const { handleItemClick } = useItemActions({
```
Comment (LuuchoRocha) [inline packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx:33]:
```suggestion
  const handleItemClick = useItemActions({
```

**Review by LuuchoRocha — CHANGES_REQUESTED** [review]

**Review by LuuchoRocha — APPROVED** [review]

**Review by leandroallemandismf — APPROVED** [review]

**Review by leandroallemandismf — APPROVED** [review]

**Review by LuuchoRocha — DISMISSED** [review]
.

**Review by tomirodeghiero — APPROVED** [review]

File: packages/ComponentLibrary/src/components/Drawer/index.tsx [hunk 4]
```diff
@@ -116,6 +124,24 @@ const Drawer: React.FC<DrawerProps> = ({ windowId, items = [], logo, title, onCl
...
+        onReportClick(reportId);
+      }
+    },
+    [onReportClick],
+  );
+
+  const handleProcessClick = useCallback(
+    (processId: string) => {
+      if (onProcessClick) {
+        onProcessClick(processId);
```
Comment (LuuchoRocha) [inline packages/ComponentLibrary/src/components/Drawer/index.tsx:139]: Is it possible to remove this callback and use onProcessClick directly, without wrapping it?

File: packages/ComponentLibrary/src/components/Drawer/index.tsx [hunk 5]
```diff
@@ -116,6 +124,24 @@ const Drawer: React.FC<DrawerProps> = ({ windowId, items = [], logo, title, onCl
     [items, onClick],
   );
 
+  const handleReportClick = useCallback(
+    (reportId: string) => {
+      if (onReportClick) {
+        onReportClick(reportId);
```
Comment (LuuchoRocha) [inline packages/ComponentLibrary/src/components/Drawer/index.tsx:130]: Is it possible to remove this callback and use onReportClick directly, without wrapping it?

File: packages/ComponentLibrary/src/components/FormView/selectors/DateSelector.tsx [hunk 6]
```diff
@@ -18,48 +18,67 @@ const StyledTextField = styled(TextField)(({ theme }) => ({
...
+        value={formatDateForInput(value)}
+        onChange={handleChange}
+        onBlur={onBlur}
+        disabled={readOnly}
+        required={required}
+        error={error}
+        helperText={helperText}
+        inputRef={inputRef}
+        InputLabelProps={{ shrink: true }}
+        InputProps={{
```
Comment (LuuchoRocha) [inline packages/ComponentLibrary/src/components/FormView/selectors/DateSelector.tsx:70]: `name` prop is missing here

File: packages/ComponentLibrary/src/components/FormView/selectors/DateSelector.tsx [hunk 7]
```diff
@@ -18,48 +18,67 @@ const StyledTextField = styled(TextField)(({ theme }) => ({
...
+        margin="normal"
+        value={formatDateForInput(value)}
+        onChange={handleChange}
+        onBlur={onBlur}
+        disabled={readOnly}
+        required={required}
+        error={error}
+        helperText={helperText}
+        inputRef={inputRef}
+        InputLabelProps={{ shrink: true }}
```
Comment (LuuchoRocha) [inline packages/ComponentLibrary/src/components/FormView/selectors/DateSelector.tsx:69]: Create a constant (out of the component) for InputLabelProps

File: packages/EtendoHookBinder/src/hooks/useReportMetadata.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    [reportId],
+  );
+
+  useEffect(() => {
+    const controller = new AbortController();
+    fetchMetadata(controller);
+    return () => {
+      controller.abort();
+    };
+  }, [reportId, fetchMetadata]);
```
Comment (LuuchoRocha) [inline packages/EtendoHookBinder/src/hooks/useReportMetadata.ts:63]: Maybe reportId is not needed here

File: packages/MainUI/app/report/[reportId]/page.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+  const handleSubmit = async (format: string, data: FieldValues) => {
+    const reportData = {
+      ...data,
+      format,
+      reportId,
+      metadata,
+    };
+    await generateReport(format, reportData);
+  };
```
Comment (LuuchoRocha) [inline packages/MainUI/app/report/[reportId]/page.tsx:36]: Move this to line 15 before the first `if` and wrap it with useCallback

File: packages/MainUI/tsconfig.json
Comment (LuuchoRocha) [inline packages/MainUI/tsconfig.json:1]: Revert this changes

**Review by LuuchoRocha — CHANGES_REQUESTED** [review]
Great job!

File: packages/ComponentLibrary/src/components/FormView/selectors/DateSelector.tsx (hunk 6 above)
Comment (Sankku) [reply packages/ComponentLibrary/src/components/FormView/selectors/DateSelector.tsx:70]: But is not?

**Pull request closed without merging**